javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/promotion/NumericConditionalExprHandler.java

```diff
@@ -0,0 +1,98 @@
+package com.github.javaparser.symbolsolver.resolution.promotion;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.github.javaparser.resolution.types.ResolvedPrimitiveType;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.resolution.typeinference.TypeHelper;
+
+/*
+ * Numeric conditional expressions are standalone expressions (§15.2).
+ * The type of a numeric conditional expression is determined as follows:
+ * If the second and third operands have the same type, then that is the type of the conditional expression.
+ * If one of the second and third operands is of primitive type T, and the type of the other is the result of applying boxing conversion (§5.1.7) to T, then the type of the conditional expression is T.
+ * If one of the operands is of type byte or Byte and the other is of type short or Short, then the type of the conditional expression is short.
+ * If one of the operands is of type T where T is byte, short, or char, and the other operand is a constant expression (§15.28) of type int whose value is representable in type T, then the type of the conditional expression is T.
+ * If one of the operands is of type T, where T is Byte, Short, or Character, and the other operand is a constant expression of type int whose value is representable in the type U which is the result of applying unboxing conversion to T, then the type of the conditional expression is U.
+ * Otherwise, binary numeric promotion (§5.6.2) is applied to the operand types, and the type of the conditional expression is the promoted type of the second and third operands.
+ */
+public class NumericConditionalExprHandler implements ConditionalExprHandler {
+
+    private static Map<ResolvedType, List<ResolvedType>> NumericConverter = new HashMap();
+    static {
+        // first type is the type of the resolution
+        // the list indicates the types that can be resolved in the type specified as
+        // key
+        NumericConverter.put(ResolvedPrimitiveType.SHORT,
+                Arrays.asList(ResolvedPrimitiveType.SHORT, ResolvedPrimitiveType.BYTE));
+    }
+
+    ResolvedType thenExpr;
+    ResolvedType elseExpr;
+
+    public NumericConditionalExprHandler(ResolvedType thenExpr, ResolvedType elseExpr) {
+        this.thenExpr = thenExpr;
+        this.elseExpr = elseExpr;
+    }
+
+    @Override
+    public ResolvedType resolveType() {
+        String qnameTypeThenExpr = thenExpr.isPrimitive() ? thenExpr.asPrimitive().describe()
```

Comment:
I think we have "ifPrimitive" too, which gives an Optional, and then you can use map().orElse()